internal/util/deepcopy.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import "reflect"
+
+// Copiable indicates that an object can be deep copied.
+type Copiable interface {
+	Copy() interface{}
+}
+
+// DeepCopyInt deep copy int. If src is nil, return nil too.
+func DeepCopyInt(src *int) *int {
+	if src == nil {
+		return (*int)(nil)
```

Comment:
why we need explicit type conversion here?

Reply:
Oh, it's not necessary. I'll remove it.